### PR TITLE
feat: add placeholder support for bucket & measurement fields

### PR DESCRIPTION
**Pull request by HongChenTW**
## Proposed Changes

Add placeholder support for bucket & measurement fields.

[Placeholder](https://docs.fluentd.org/configuration/buffer-section#placeholders) is a very useful feature provided by fluentd, we can leverage this feature to dynamically specify measurement & bucket parameters.

## Checklist

- [x] CHANGELOG.md updated
- [x] Rebased/mergeable
- [x] A test has been added if appropriate
- [x] `rake test` completes successfully
- [x] Commit messages are in [semantic format](https://seesparkbox.com/foundry/semantic_commit_messages)
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: lib/fluent/plugin/out_influxdb2.rb [hunk 1]
```diff
@@ -141,12 +141,24 @@ def write(chunk)
...
+
+    if @measurement
+      measurement = extract_placeholders(@measurement, chunk)
+    else
+      measurement = chunk.metadata.tag
+    end
+
+    return bucket, measurement
+  end
+
```
Comment (bednar) [inline lib/fluent/plugin/out_influxdb2.rb:161]: Please use correct code style:

```suggestion
  def expand_placeholders(chunk)
    bucket = extract_placeholders(@bucket, chunk)

    measurement = if @measurement
                    extract_placeholders(@measurement, chunk)
                  else
                    chunk.metadata.tag
                  end

    [bucket, measurement]
  end
```

You can test code formatting by: `bundle exec rake rubocop`

**Review by bednar — CHANGES_REQUESTED** [review]
Thanks for your PR 👍 
There is a few requirements that should be satisfy before we accept the PR:

1. Please update `CHANGELOG.md`
2. Please add test for your changes, something like:
```ruby
def test_measurement_as_placeholder
    stub_request(:any, 'https://localhost:8086/api/v2/write?bucket=my-bucket&org=my-org&precision=ns')
      .to_return(status: 204)
    driver = create_driver(%(
      @type influxdb2
      token my-token
      bucket my-bucket
      org my-org
      time_precision ns
      measurement placeholder_${tag}
    ))
    driver.run(default_tag: 'h2o_tag') do
      emit_documents(driver)
    end
    assert_requested(:post, 'https://localhost:8086/api/v2/write?bucket=my-bucket&org=my-org&precision=ns',
                     times: 1, body: 'placeholder_h2o_tag level=2i,location="europe" 1444897215000000000')
end
```

and also:

File: lib/fluent/plugin/out_influxdb2.rb (hunk 1 above)
Comment (HongChenTW) [reply lib/fluent/plugin/out_influxdb2.rb]: Thanks for the suggestion, I've update the code style.

**Comment by HongChenTW** [comment]
Hi @bednar, thanks for the informations, I've updated all needed tasks in checklist, but I forgot to add semantic commit messages, feel free to squash the commits.

**Review by bednar — APPROVED** [review]
Thanks again for your PR 👍.

LGTM

**Pull request merged**